src/webui/www/private/scripts/prop-files.js

```diff
@@ -173,7 +173,7 @@ window.qBittorrent.PropFiles = (function() {
         elem.set('value', priority.toString());
         elem.set('html', html);
         if (selected)
-            elem.setAttribute('selected', '');
+            elem.setProperty('selected', true);
```

Comment:
Could you test with:
```suggestion
            elem.setAttribute('selected', 'true');
```

`setProperty` underneath still calls `setAttribute`: https://github.com/mootools/mootools-core/blob/master/Source/Element/Element.js#L707-L730 so it doesn't do anything magic.

Reply:
Was it hitting a Safari bug?
`selected` is a boolean attribute: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/option
and from https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute#examples:
>To set the value of a Boolean attribute, such as disabled, you can specify any value. **An empty string or the name of the attribute are recommended values.** All that matters is that if the attribute is present at all, regardless of its actual value, its value is considered to be true. The absence of the attribute means its value is false.

I don't mind we accept workarounds and I mean you could also considering reporting the bug to Safari.

Reply:
> Was it hitting a Safari bug?

Safari cannot contain a bug (Apple). 😁

Reply:
> Could you test with: `elem.setAttribute('selected', true);`

This did not work. Same as an empty value.

> `setProperty` underneath still calls `setAttribute`: https://github.com/mootools/mootools-core/blob/master/Source/Element/Element.js#L707-L730 so it doesn't do anything magic.

That's not quite true as far as I understand it. `setProperty` checks if the `name` exists in `propertySetters` and if it does a different function will be called. The boolean attributes [are added to `propertySetters`](https://github.com/mootools/mootools-core/blob/187a16bae2d7144ea4b81aa5c0586498f0fe17c7/Source/Element/Element.js#L577-L596) with a different function that is not using `setAttribute` but rather setting a property on the DOM.

> Was it hitting a Safari bug?

As far as I understand it it's not really a bug but [rather a feature of how the web browsers are working with attributes and properties](https://javascript.info/dom-attributes-and-properties). I do however agree that it is unexpected behavior, especially since the other web browsers are creating the expected results.

For consistency I would suggest replacing all occurrences of `setAttribute` in the code with `setProperty`, though I have not done that in this pull request. Using `setProperty` should prevent future subtile bugs without any downsides since MooTools will still call `setAttribute` for all non-boolean attributes.

Reply:
>That's not quite true as far as I understand it. setProperty checks if the name exists in propertySetters and if it does a different function will be called. 

You are right here.

>For consistency I would suggest replacing all occurrences of setAttribute in the code with setProperty, though I have not done that in this pull request. Using setProperty should prevent future subtile bugs without any downsides since MooTools will still call setAttribute for all non-boolean attributes.

mootools is not maintained anymore and we would like to rely less on it. What about setting the property directly such as `elem.selected='';` or `elem.selected='true';`?

>As far as I understand it it's not really a bug but [rather a feature of how the web browsers are working with attributes and properties](https://javascript.info/dom-attributes-and-properties). I do however agree that it is unexpected behavior, especially since the other web browsers are creating the expected results.

Ehh... unexpected behavior yet not a bug, this is what causes the web environment to be fragmented.